legend-depot-core-http/src/main/java/org/finos/legend/depot/tracing/resources/BaseResource.java

```diff
@@ -94,12 +98,37 @@ protected <T> T handle(String resourceAPIMetricName, String label, Supplier<T> s
         return TracerFactory.get().executeWithTrace(label, () -> handleWithLogging(resourceAPIMetricName, label, supplier));
     }
 
+    protected <T> Response handle(String resourceAPIMetricName, String label, Supplier<T> supplier, Request request, Supplier<EntityTag> etagSupplier)
+    {
+        EntityTag serverTag;
+        try
+        {
+            serverTag = etagSupplier.get();
+        }
+        catch (Exception e)
+        {
+            serverTag = null;
+        }
+        if (serverTag != null && request != null && request.evaluatePreconditions(serverTag) != null)
+        {
+            return Response.noContent().status(Response.Status.NOT_MODIFIED).build();
+        }
+        try
+        {
+            T output = TracerFactory.get().executeWithTrace(label, () -> handleWithLogging(resourceAPIMetricName, label, supplier));
+            return Response.ok(output).tag(serverTag).build();
+        }
+        catch (Exception e)
```

Comment:
do you really need the try catch...if there is an exception if will be dealt in the hadlewithloggin method.. u dont needto override that behavour